--- a/Sources/MarkdownUI/Views/Blocks/BlockSequence.swift
+++ b/Sources/MarkdownUI/Views/Blocks/BlockSequence.swift
@@ -24,10 +24,10 @@
 
   var body: some View {
     VStack(alignment: self.textAlignment.alignment.horizontal, spacing: 0) {
-      ForEach(self.data, id: \.self) { element in
+      ForEach(self.data, id: \.index) { element in
         self.content(element.index, element.value)
           .onPreferenceChange(BlockMarginsPreference.self) { value in
-            self.blockMargins[element.hashValue] = value
+            self.blockMargins[element.index] = value
           }
           .padding(.top, self.topPaddingLength(for: element))
       }
@@ -39,10 +39,10 @@
       return 0
     }
 
-    let topSpacing = self.blockMargins[element.hashValue]?.top
+    let topSpacing = self.blockMargins[element.index]?.top
     let predecessor = self.data[element.index - 1]
     let predecessorBottomSpacing =
-      self.tightSpacingEnabled ? 0 : self.blockMargins[predecessor.hashValue]?.bottom
+      self.tightSpacingEnabled ? 0 : self.blockMargins[predecessor.index]?.bottom
 
     return [topSpacing, predecessorBottomSpacing]
       .compactMap { $0 }
--- a/Sources/MarkdownUI/Views/Blocks/ListItemView.swift
+++ b/Sources/MarkdownUI/Views/Blocks/ListItemView.swift
@@ -37,6 +37,7 @@
       self.markerStyle
         .makeBody(configuration: .init(listLevel: self.listLevel, itemNumber: self.number))
         .textStyleFont()
+        .fixedSize(horizontal: true, vertical: false)
         .readWidth(column: 0)
         .frame(width: self.markerWidth, alignment: .trailing)
     }
